src/sonic-nettools/wol/src/wol.rs

```diff
@@ -0,0 +1,687 @@
+    fn test_is_mac_string_valid() {
+        assert!(is_mac_string_valid("00:11:22:33:44:55"));
+        assert!(!is_mac_string_valid(""));
+        assert!(!is_mac_string_valid("0:1:2:3:4:G"));
+        assert!(!is_mac_string_valid("00:11:22:33:44:GG"));
+        assert!(!is_mac_string_valid("00-11-22-33-44-55"));
+        assert!(!is_mac_string_valid("00:11:22:33:44:55:66"));
+    }
+
+    #[test]
+    fn test_is_ipv4_address_valid() {
+        assert!(is_ipv4_address_valid("192.168.1.1"));
+        assert!(!is_ipv4_address_valid(""));
+        assert!(!is_ipv4_address_valid("0::1"));
+        assert!(!is_ipv4_address_valid("192.168.1"));
+        assert!(!is_ipv4_address_valid("192.168.1.256"));
+        assert!(!is_ipv4_address_valid("192.168.1.1.1"));
+    }
+
+    #[test]
+    fn test_build_magic_packet() {
+        let src_mac = [0x00, 0x11, 0x22, 0x33, 0x44, 0x55];
+        let target_mac = [0xff, 0xff, 0xff, 0xff, 0xff, 0xff];
+        let four_bytes_password = Some(Password(vec![0x00, 0x11, 0x22, 0x33]));
+        let magic_packet =
+            build_magic_packet(&src_mac, &target_mac, &target_mac, &four_bytes_password).unwrap();
+        assert_eq!(magic_packet.len(), 120);
+        assert_eq!(&magic_packet[0..6], &target_mac);
+        assert_eq!(&magic_packet[6..12], &src_mac);
+        assert_eq!(&magic_packet[12..14], &[0x08, 0x42]);
+        assert_eq!(&magic_packet[14..20], &[0xff; 6]);
+        assert_eq!(&magic_packet[20..116], target_mac.repeat(16));
+        assert_eq!(&magic_packet[116..120], &[0x00, 0x11, 0x22, 0x33]);
+        let six_bytes_password = Some(Password(vec![0x00, 0x11, 0x22, 0x33, 0x44, 0x55]));
+        let magic_packet =
+            build_magic_packet(&src_mac, &target_mac, &target_mac, &six_bytes_password).unwrap();
+        assert_eq!(magic_packet.len(), 122);
+        assert_eq!(&magic_packet[0..6], &target_mac);
+        assert_eq!(&magic_packet[6..12], &src_mac);
+        assert_eq!(&magic_packet[12..14], &[0x08, 0x42]);
+        assert_eq!(&magic_packet[14..20], &[0xff; 6]);
+        assert_eq!(&magic_packet[20..116], target_mac.repeat(16));
+        assert_eq!(
+            &magic_packet[116..122],
+            &[0x00, 0x11, 0x22, 0x33, 0x44, 0x55]
+        );
+    }
+
+    #[test]
+    fn test_build_magic_packet_without_password() {
+        let src_mac = [0x00, 0x11, 0x22, 0x33, 0x44, 0x55];
+        let dst_mac = [0xff, 0xff, 0xff, 0xff, 0xff, 0xff];
+        let target_mac = [0x01, 0x02, 0x03, 0x04, 0x05, 0x06];
+        let magic_packet = build_magic_packet(&src_mac, &dst_mac, &target_mac, &None).unwrap();
+        assert_eq!(magic_packet.len(), 116);
+        assert_eq!(&magic_packet[0..6], &dst_mac);
+        assert_eq!(&magic_packet[6..12], &src_mac);
+        assert_eq!(&magic_packet[12..14], &[0x08, 0x42]);
+        assert_eq!(&magic_packet[14..20], &[0xff; 6]);
+        assert_eq!(&magic_packet[20..116], target_mac.repeat(16));
+    }
+
+    #[test]
+    fn verify_args_parse() {
```

Comment:
It's better to split this testcase to several testcases, each one for a single argument or scenario.